### PR TITLE
Feature/3253/electron swap

### PR DESCRIPTION
**Pull request by phanlezz**
# Swap to a new runner for Visualization (From NW.JS to Electron)

Due to the issues with nw.js and its builder derivatives we want to swap to electron for the visualization.
This means, that we also swap to a new packager.

Additionally, electron provides an arm version which enables codecharta to run natively on new apple silicon.

Issue: #3253 

## Workflow

- [x] Swap to electron as runner `npm run start`
- [x] Swap to  an electron packager for packaging of the standalone version `npm run package`
- [x] Write an ADR

## Notes
Standalone desktop app tested on Windows 10/11 and Ubuntu 22.04.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 
 ### Chore 👨‍💻 👩‍💻
 
--   Raised minimum required Java Version from 8 to 11 [#3357](https://github.com/MaibornWolff/codecharta/issues/3357)
+-   Raise minimum required Java Version from 8 to 11 [#3357](https://github.com/MaibornWolff/codecharta/issues/3357)
+-   Swap nw.js to electron for standalone execution [#3253](https://github.com/MaibornWolff/codecharta/issues/3253)
```
Comment (ce-bo) [inline CHANGELOG.md:13]: Please link the PR instead of the Issue.

File: gh-pages/_posts/adr/2023-10-09-ADR_10_change_visualization_standalone.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# Status
+
+accepted 1.
+
+# Decision
+
+1. Use the _electron_ framework for standalone CodeCharta-Visualization
+2. Use the [electron-packager](https://github.com/electron/electron-packager) to produce all available combinations of OS and architectures
+3. Zip the results like previously and add them as assets to releases.
```
Comment (ce-bo) [inline gh-pages/_posts/adr/2023-10-09-ADR_10_change_visualization_standalone.md:38]: I think it is both, a decision and a consequence.

File: visualization/package.json [hunk 3]
```diff
@@ -70,8 +65,6 @@
 	"dependencies": {
 		"@angular/animations": "^15.1.1",
 		"@angular/cdk": "^15.1.1",
-		"@angular/common": "^15.1.1",
-		"@angular/core": "^15.1.1",
```
Comment (ce-bo) [inline visualization/package.json:74]: We don't need them anymore?

**Review by ce-bo — APPROVED** [review]
In general LGTM. Before we release that let's do a test session.

File: visualization/package.json (hunk 3 above)
Comment (phanlezz) [reply visualization/package.json:74]: Good question, I'm not sure why they got removed. I will look this.

File: CHANGELOG.md (hunk 1 above)
Comment (phanlezz) [reply CHANGELOG.md:13]: Woops😆

**Pull request merged**